x-pack/plugin/autoscaling/src/internalClusterTest/java/org/elasticsearch/xpack/autoscaling/action/GetAutoscalingCapacityRestCancellationIT.java

```diff
@@ -0,0 +1,147 @@
+import static org.hamcrest.Matchers.equalTo;
+
+@ESIntegTestCase.ClusterScope(scope = ESIntegTestCase.Scope.TEST, numDataNodes = 0)
+public class GetAutoscalingCapacityRestCancellationIT extends AutoscalingIntegTestCase {
+
+    @Override
+    protected boolean addMockHttpTransport() {
+        return false; // enable http
+    }
+
+    @Override
+    protected Collection<Class<? extends Plugin>> nodePlugins() {
+        List<Class<? extends Plugin>> result = new ArrayList<>(super.nodePlugins());
+        result.add(Netty4Plugin.class);
+        return Collections.unmodifiableList(result);
+    }
+
+    @Override
+    protected boolean ignoreExternalCluster() {
+        return true;
+    }
+
+    public void testCapacityRestCancellationAndResponse() throws Exception {
+        internalCluster().startMasterOnlyNode();
+
+        putAutoscalingPolicy(Map.of("count", Settings.EMPTY, "!sync", Settings.builder().put("check_for_cancel", randomBoolean()).build()));
+        try (RestClient restClient = createRestClient()) {
+
+            PlainActionFuture<Response> cancelledFuture = new PlainActionFuture<>();
+            PlainActionFuture<Response> successFuture1 = new PlainActionFuture<>();
+            PlainActionFuture<Response> successFuture2 = new PlainActionFuture<>();
+            Request getCapacityRequest = new Request("GET", "/_autoscaling/capacity");
+            Cancellable cancellable = restClient.performRequestAsync(getCapacityRequest, wrapAsRestResponseListener(cancelledFuture));
+            LocalStateAutoscaling.AutoscalingTestPlugin plugin = internalCluster().getMasterNodeInstance(PluginsService.class)
+                .filterPlugins(LocalStateAutoscaling.class)
+                .get(0)
+                .testPlugin();
+            plugin.syncWithDeciderService(() -> {
+                putAutoscalingPolicy(Map.of("count", Settings.EMPTY));
+                assertThat(
+                    internalCluster().getMasterNodeInstance(TransportGetAutoscalingCapacityAction.class).responseCacheQueueSize(),
+                    equalTo(1)
+                );
+                restClient.performRequestAsync(getCapacityRequest, wrapAsRestResponseListener(successFuture1));
+                assertBusy(
+                    () -> assertThat(
+                        internalCluster().getMasterNodeInstance(TransportGetAutoscalingCapacityAction.class).responseCacheQueueSize(),
+                        equalTo(2)
+                    )
+                );
+                restClient.performRequestAsync(getCapacityRequest, wrapAsRestResponseListener(successFuture2));
+                assertBusy(
+                    () -> assertThat(
+                        internalCluster().getMasterNodeInstance(TransportGetAutoscalingCapacityAction.class).responseCacheQueueSize(),
+                        equalTo(3)
+                    )
+                );
+                cancellable.cancel();
+                waitForCancelledCapacityTask();
+            });
```

Comment:
Can we document that the other tasks are still blocked here?

```suggestion

                assertFalse(successFuture1.isDone());
                assertFalse(successFuture2.isDone());
            });
```

Reply:
:thu: cf9d5d4013f8eb4ff9a03990a7144669ba4c2fd4